### PR TITLE
#3 #18 Abstract Items

### PR DESCRIPTION
**Pull request by mitsudz**
Creation of item interface and implementation within abstract item class. 

Branch used for feature #3  and task #18

### CODE REVIEW

**Review by PHONGNGYUEN1 — APPROVED** [review]
This looks pretty good I approve request

File: source/core/src/main/com/csse3200/game/inventory/items/ItemInterface.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
+package com.csse3200.game.inventory.items;
+
+/**
+ * Represents an item that can be used by a player and stored within the inventory.
+ */
```
Comment (lellis0406) [inline source/core/src/main/com/csse3200/game/inventory/items/ItemInterface.java:5]: ItemInterface class is redundant, just use AbstractItem instead

File: source/core/src/main/com/csse3200/game/inventory/items/AbstractItem.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    @Override
+    public int getQuantity() {return quantity;}
+
+    /**
+     * Checks the number of this item that can be further stacked.
+     *
+     * @return how much the quantity of this item can increase by
+     */
+    @Override
+    public int canAdd() {return limit - quantity;}
```
Comment (lellis0406) [inline source/core/src/main/com/csse3200/game/inventory/items/AbstractItem.java:97]: naming implies boolean return value

**Review by lellis0406 — CHANGES_REQUESTED** [review]
I've reviewed and left some minor adjustment and redundancies to be addressed. Otherwise looks good.

File: source/core/src/main/com/csse3200/game/inventory/items/AbstractItem.java (hunk 2 above)
Comment (mitsudz) [reply source/core/src/main/com/csse3200/game/inventory/items/AbstractItem.java]: resolved (changed name from canAdd to numAddable)

**Pull request merged**